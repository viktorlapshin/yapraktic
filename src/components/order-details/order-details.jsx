import React from "react";
import styles from "./order-details.module.css";
import { CloseIcon } from "@ya.praktikum/react-developer-burger-ui-components";

export const OrderDetails = ({ onClose }) => {
  return (
    <>
      <div className={styles.order_block}>
        <button
          className={styles.close_button}
          onClick={onClose}
          aria-label="Закрыть"
        >
          <CloseIcon />
        </button>
        <p className={styles.order_number}>034536</p>
        <p className={`${styles.title} text text_type_main-large`}>
          идентификатор заказа
        </p>
        <img className={styles.check_mark} src="./public/images/check_mark.png" alt="check_mark" />
        <p className={`${styles.title} text text_type_main`}>
          Ваш заказ начали готовить
        </p>
        <p className={styles.modal_text}>
          Дождитесь готовности на орбитальной станции
        </p>
      </div>
    </>
  );
};
