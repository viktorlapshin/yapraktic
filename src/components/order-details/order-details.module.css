.order_block {
  display: flex;
  width: 720px;
  height: 718px;
  flex-direction: column;
  align-items: center;
  gap: 40px;
}

.order_number {
  width: 520px;
  height: 120px;
  font-size: 160px;
  font-family: Iceland;
  font-weight: 200;
  margin: 0;
  text-align: center;
}

.check_mark {
  width: 120px;
  height: 120px;
}

.modal_text {
  margin: 0;
  color: #8585ad;
}

.close_button {
  width: 640px;
  background: transparent;
  border: none;
  display: flex;
  justify-content: flex-end;
  margin-top: 60px;
}
