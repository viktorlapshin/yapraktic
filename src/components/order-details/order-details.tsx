import styles from "./order-details.module.css";
import { CloseIcon } from "@ya.praktikum/react-developer-burger-ui-components";
import { FC } from "react";

interface OrderDetailsProps {
  orderNumber: number | null;
  onClose: () => void;
}

export const OrderDetails: FC<OrderDetailsProps> = ({
  orderNumber,
  onClose,
}) => {
  return (
    <div className={styles.order_block}>
      <button
        className={styles.close_button}
        onClick={onClose}
        aria-label="Закрыть"
        type="button"
        data-testid="close-button"
      >
        <CloseIcon type="primary" />
      </button>
      <p className={styles.order_number} data-cy="order-details-number">
        {orderNumber ? orderNumber : "—"}
      </p>
      <p className={`${styles.title} text text_type_main-large`}>
        идентификатор заказа
      </p>
      <img
        className={styles.check_mark}
        src="/images/check_mark.png"
        alt="check_mark"
      />
      <p className={`${styles.title} text text_type_main`}>
        Ваш заказ начали готовить
      </p>
      <p className={styles.modal_text}>
        Дождитесь готовности на орбитальной станции
      </p>
    </div>
  );
};
