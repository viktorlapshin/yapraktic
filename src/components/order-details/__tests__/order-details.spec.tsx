import { describe } from "@jest/globals";
import { render, screen } from "@testing-library/react";
import { OrderDetails } from "../order-details";

describe("Header component", () => {
  it("should be equal to snapshot", () => {
    const { asFragment } = render(
      <OrderDetails onClose={() => {}} orderNumber={123} />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("test close click", () => {
    let isClicked = false;

    render(
      <OrderDetails
        onClose={() => {
          isClicked = true;
        }}
        orderNumber={123}
      />
    );

    const element = screen.getByTestId("close-button");

    element.click();

    expect(isClicked).toBeTruthy();
  });
});
