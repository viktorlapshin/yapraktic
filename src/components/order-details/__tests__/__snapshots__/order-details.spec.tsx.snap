// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Header component should be equal to snapshot 1`] = `
<DocumentFragment>
  <div
    class="order_block"
  >
    <button
      aria-label="Закрыть"
      class="close_button"
      data-testid="close-button"
      type="button"
    >
      <svg
        fill="#F2F2F3"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M3.29289 3.29289C3.68342 2.90237 4.31658 2.90237 4.70711 3.29289L12 10.5858L19.2929 3.29289C19.6834 2.90237 20.3166 2.90237 20.7071 3.29289C21.0976 3.68342 21.0976 4.31658 20.7071 4.70711L13.4142 12L20.7071 19.2929C21.0976 19.6834 21.0976 20.3166 20.7071 20.7071C20.3166 21.0976 19.6834 21.0976 19.2929 20.7071L12 13.4142L4.70711 20.7071C4.31658 21.0976 3.68342 21.0976 3.29289 20.7071C2.90237 20.3166 2.90237 19.6834 3.29289 19.2929L10.5858 12L3.29289 4.70711C2.90237 4.31658 2.90237 3.68342 3.29289 3.29289Z"
          fill-rule="evenodd"
        />
      </svg>
    </button>
    <p
      class="order_number"
      data-cy="order-details-number"
    >
      123
    </p>
    <p
      class="title text text_type_main-large"
    >
      идентификатор заказа
    </p>
    <img
      alt="check_mark"
      class="check_mark"
      src="/images/check_mark.png"
    />
    <p
      class="title text text_type_main"
    >
      Ваш заказ начали готовить
    </p>
    <p
      class="modal_text"
    >
      Дождитесь готовности на орбитальной станции
    </p>
  </div>
</DocumentFragment>
`;
